arch/posix/CMakeLists.txt

```diff
@@ -1,13 +1,13 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if (CONFIG_64BIT)
-	# some gcc versions fail to build without -fPIC
-	zephyr_compile_options(-m64 -fPIC)
-	zephyr_link_libraries(-m64)
-else ()
-	zephyr_compile_options(-m32)
-	zephyr_link_libraries(-m32)
-endif ()
+# @Intent: Set necessary compiler & linker options that apply globally for OS
+include(${CMAKE_HOST_SYSTEM_NAME}.cmake OPTIONAL)
+
+# @Intent: Set necessary compiler & linker options that apply globally for ARCH
+include(${CMAKE_HOST_SYSTEM_PROCESSOR}.cmake OPTIONAL)
+
+# @Intent: Set necessary compiler & linker options for the specific OS-ARCH tuple
+include(${CMAKE_HOST_SYSTEM_NAME}.${CMAKE_HOST_SYSTEM_PROCESSOR}.cmake OPTIONAL)
```

Comment:
This is really changing current behavior.
Have you ensured this will work as expected on all platforms / boards ?
One thing is to ensure we are not breaking officially supported platforms, but there are still platforms / toolchains which are not officially supported, but still works, and people are using.

Also, this means each platform needs a file, even if the content is identical to another systems file.
But it is also nice to split into logical groups, so maybe something like:
```suggestion
if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/${CMAKE_HOST_SYSTEM_NAME}.${CMAKE_HOST_SYSTEM_PROCESSOR}.cmake)
  # @Intent: Set necessary compiler & linker options for the specific OS-ARCH tuple
  include(${CMAKE_HOST_SYSTEM_NAME}.${CMAKE_HOST_SYSTEM_PROCESSOR}.cmake)
else()
  if (CONFIG_64BIT)
    # some gcc versions fail to build without -fPIC
    zephyr_compile_options(-m64 -fPIC)
    zephyr_link_libraries(-m64)
  else ()
    zephyr_compile_options(-m32)
    zephyr_link_libraries(-m32)
  endif ()
endif()
```

that way we ensure current behavior is not changing, while allowing the flexibility of placing platform specific files.
Let's wait with extending with `${CMAKE_HOST_SYSTEM_PROCESSOR}.cmake` and `${CMAKE_HOST_SYSTEM_NAME}.cmake` until needed.
We should in general try to minimize host dependent variations, although it is needed in some case, like this PR shows.

Reply:
See previous [comment](https://github.com/zephyrproject-rtos/zephyr/pull/24843#discussion_r422088474)